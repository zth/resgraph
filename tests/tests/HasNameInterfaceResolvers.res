open Interface_hasName

@gql.field
let hasName = (_: Query.query, ~id: ResGraph.id): option<Resolver.t> => {
  ignore(id)
  Some(User({id: "123", name: "Test", lastAge: Some(35), age: 35}))
}

@gql.field
let abbreviatedName = (
  // ^hov
  name: HasNameInterface.hasName,
  ~typeName: ImplementedBy.t,
) => {
  Some(typeName->ImplementedBy.toString ++ ":" ++ name.name)
}
