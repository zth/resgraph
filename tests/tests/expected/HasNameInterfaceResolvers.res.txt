Hover tests/HasNameInterfaceResolvers.res 9:5
{"status": "Hover", "item": {"contents": {"kind": "markdown", "value": "GraphQL interface field generated by ResGraph:\n```graphql\ninterface HasName {\n  abbreviatedName: String\n}\n```"}}}
